docs/src/references/agent/software-management.md

```diff
@@ -0,0 +1,376 @@
+      - optionally a `"version"` (using the same conventions as the package packager),
+      - optionally an `"url"` from where to download the package.
+
+As an example, here is a message requesting a `software-update` on a child device:
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update/c8y-2023-09-25T14:53:00' '{
+    "status": "init",
+    "updateList": [
+        {
+            "type": "apt",
+            "modules": [
+                {
+                    "name": "nodered",
+                    "version": "1.0.0",
+                    "action": "install"
+                },
+                {
+                    "name": "collectd",
+                    "version": "5.12",
+                    "url": "https://collectd.org/download/collectd-tarballs/collectd-5.12.0.tar.bz2",
+                    "action": "install"
+                }
+            ]
+        },
+        {
+            "type": "docker",
+            "modules": [
+                {
+                    "name": "nginx",
+                    "version": "1.21.0",
+                    "action": "install"
+                },
+                {
+                    "name": "mongodb",
+                    "version": "4.4.6",
+                    "action": "remove"
+                }
+            ]
+        }
+    ]
+}'
+```
+
+### `executing` state
+
+Just before starting the command execution, the agent marks the command as executing
+by publishing a retained message that repeats the former command except that:
+
+- the `status` is set to `executing`
+
+### `successful` state
+
+The payload for a successful `software-update` command
+repeats the same content as the former request except that:
+
+- the `status` is set to `successful`.
+
+As an example, here is a status message for a successful `software-update` command on a child device:
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update/c8y-2023-09-25T14:53:00' '{
+    "status": "successful",  
+    "updateList": [
+        {
+            "type": "apt",
+            "modules": [
+                {
+                    "name": "nodered",
+                    "version": "1.0.0",
+                    "action": "install"
+                },
+                {
+                    "name": "collectd",
+                    "version": "5.12",
+                    "url": "https://collectd.org/download/collectd-tarballs/collectd-5.12.0.tar.bz2",
+                    "action": "install"
+                }
+            ]
+        },
+        {
+            "type": "docker",
+            "modules": [
+                {
+                    "name": "nginx",
+                    "version": "1.21.0",
+                    "action": "install"
+                },
+                {
+                    "name": "mongodb",
+                    "version": "4.4.6",
+                    "action": "remove"
+                }
+            ]
+        }
+    ]
+}'
```

Comment:
I'm assuming that the omission of the `currentSoftwareList` from the update command response intentional. I'm juts curious why. Of course, the mapper can always trigger the `software_list` command immediately after it gets the `successful` response from the agent, to report the up-to-date software list to the cloud. The `currentSoftwareList` was included earlier just to avoid this extra hop.